qcodes/instrument_drivers/Keysight/N52xx.py

```diff
@@ -0,0 +1,436 @@
+
+        # Number of points in a sweep
+        self.add_parameter('points',
+                           label='Points',
+                           get_cmd='SENS:SWE:POIN?',
+                           get_parser=int,
+                           set_cmd='SENS:SWE:POIN {}',
+                           unit='',
+                           vals=Numbers(min_value=1, max_value=100001))
+
+        # Electrical delay
+        self.add_parameter('electrical_delay',
+                           label='Electrical Delay',
+                           get_cmd='CALC:CORR:EDEL:TIME?',
+                           get_parser=float,
+                           set_cmd='CALC:CORR:EDEL:TIME {:.6e}',
+                           unit='s',
+                           vals=Numbers(min_value=0, max_value=100000))
+
+        # Sweep Time
+        self.add_parameter('sweep_time',
+                           label='Time',
+                           get_cmd='SENS:SWE:TIME?',
+                           get_parser=float,
+                           unit='s',
+                           vals=Numbers(0,1e6))
+        # Sweep Mode
+        self.add_parameter('sweep_mode',
+                           label='Mode',
+                           get_cmd='SENS:SWE:MODE?',
+                           set_cmd='SENS:SWE:MODE {}',
+                           vals=Enum("HOLD", "CONT", "GRO", "SING"))
+
+        # Traces
+        # Note: These will be accessed through the traces property which updates
+        # the channellist to include only active trace numbers
+        self._traces = ChannelList(self, "PNATraces", PNATrace)
+        self.add_submodule("traces", self._traces)
+        # Add shortcuts to trace 1
+        trace1 = PNATrace(self, "tr1", 1)
+        for param in trace1.parameters.values():
+            self.parameters[param.name] = param
+        # By default we should also pull any following values from this trace
+        self.write("CALC:PAR:MNUM 1")
+
+        # Set auto_sweep parameter
+        # If we want to return multiple traces per setpoint without sweeping
+        # multiple times, we should set this to false
+        self._auto_sweep = True
+        self.add_parameter('auto_sweep',
+                           label='Auto Sweep',
+                           set_cmd=self._set_auto_sweep,
+                           get_cmd=lambda: self._auto_sweep,
+                           vals=Bool())
+
+        # Functions
+        # Clear averages
+        self.add_function('reset_averages',
+                          call_cmd='SENS:AVER:CLE')
+        # Averages ON
+        self.add_function('averages_on',
+                          call_cmd='SENS:AVER ON')
+        # Averages OFF
+        self.add_function('averages_off',
+                          call_cmd='SENS:AVER OFF')
+
+        # A default output format on initialisation
+        self.write('FORM REAL,32')
+        self.write('FORM:BORD NORM')
```

Comment:
are these two commands not supposed to be called elsewhere/elsewhen? because if they might, please, make each of them a method (or a parameter, perhaps)?

Reply:
This just sets the transfer format, I can't see any reason why they need to be called again or changed once set. There is no way to change their values in the UI, so it only needs to happen once when connecting...

Reply:
ok